docs/02-app/01-building-your-application/11-upgrading/02-version-15.mdx

```diff
@@ -7,39 +7,448 @@ description: Upgrade your Next.js Application from Version 14 to 15.
 
 ## Upgrading from 14 to 15
 
-To update to Next.js version 15, run the following command using your preferred package manager:
+To update to Next.js version 15, you can use the `upgrade` codemod:
 
 ```bash filename="Terminal"
-npm i next@rc react@rc react-dom@rc eslint-config-next@rc
+npx @next/codemod@canary upgrade
 ```
 
-```bash filename="Terminal"
-yarn add next@rc react@rc react-dom@rc eslint-config-next@rc
-```
-
-```bash filename="Terminal"
-pnpm up next@rc react@rc react-dom@rc eslint-config-next@rc
-```
+If you prefer to do it manually, ensure that you're installing the latest Next & React RC, e.g.:
 
 ```bash filename="Terminal"
-bun add next@rc react@rc react-dom@rc eslint-config-next@rc
+npm i next@canary react@rc react-dom@rc eslint-config-next@rc
 ```
 
 > **Good to know:**
 >
 > - If you see a peer dependencies warning, you may need to update `react` and `react-dom` to the suggested versions, or you use the `--force` or `--legacy-peer-deps` flag to ignore the warning. This won't be necessary once both Next.js 15 and React 19 are stable.
 > - If you are using TypeScript, you'll need to temporarily override the React types. See the [React 19 RC upgrade guide](https://react.dev/blog/2024/04/25/react-19-upgrade-guide#installing) for more information.
 
-## Minimum React version
-
-- The minimum `react` and `react-dom` is now 19.
-
 ## React 19
 
+- The minimum versions of `react` and `react-dom` is now 19.
 - `useFormState` has been replaced by `useActionState`. The `useFormState` hook is still available in React 19, but it is deprecated and will be removed in a future release. `useActionState` is recommended and includes additional properties like reading the `pending` state directly. [Learn more](https://react.dev/reference/react/useActionState).
 - `useFormStatus` now includes additional keys like `data`, `method`, and `action`. If you are not using React 19, only the `pending` key is available. [Learn more](https://react.dev/reference/react-dom/hooks/useFormStatus).
 - Read more in the [React 19 upgrade guide](https://react.dev/blog/2024/04/25/react-19-upgrade-guide).
 
+## Async Request APIs (Breaking change)
+
+Previously synchronous Dynamic APIs that rely on runtime information are now **asynchronous**:
+
+- [`cookies`](/docs/app/api-reference/functions/cookies)
+- [`headers`](/docs/app/api-reference/functions/headers)
+- [`draftMode`](/docs/app/api-reference/functions/draft-mode)
+- `params` in [`layout.js`](/docs/app/api-reference/file-conventions/layout), [`page.js`](/docs/app/api-reference/file-conventions/page), [`route.js`](/docs/app/api-reference/file-conventions/route), and [`default.js`](/docs/app/api-reference/file-conventions/default).
+- `searchParams` in [`page.js`](/docs/app/api-reference/file-conventions/page)
+
+To ease the burden of migration, a [codemod is available](/docs/app/building-your-application/upgrading/codemods#150) to automate the process and the APIs can temporarily be accessed synchronously.
+
```

Comment:
do we also wanna have a link to `errors/sync-dynamic-apis`? Later we could add more details there

Reply:
I think it probably makes sense to link to that in code and not in documentation as it's about a very specific warning.